src/Parser/DeathKnight/Blood/Modules/Talents/FoulBulwark.js

```diff
@@ -27,7 +27,7 @@ class FoulBulwark extends Analyzer {
   get averageFoulBullwark() {
     let avgStacks = 0;
     this.boneShieldTimesByStack.forEach((elem, index) => {
-      avgStacks += elem.reduce((a, b) => a + b) / this.owner.fightDuration * index;
+      avgStacks += elem.length > 0 ? elem.reduce((a, b) => a + b) / this.owner.fightDuration * index : 0;
```

Comment:
The error is caused by the reducer not having a default value, I think giving it a default value might lead to simpler code.

Reply:
Definitely cleaner, updated it